apigentools/utils.py

```diff
@@ -130,43 +146,58 @@ def run_command(cmd, log_level=logging.INFO, additional_env=None, combine_out_er
     :param dry_run: Whether or not this is a dry run (``True``, commands are not executed, just logged)
         or real run (``False``)
     :type dry_run: ``bool``
+    :param sensitive_output: Whether or not the output of the called subprocess is sensitive or not.
+        If true, all logging will be suppressed and if a subprocess.CalledProcessError is raised,
+        its attributes will be empty
+    :type sensitive_output: ``bool``
     :return: Result of the called subprocess
     :rtype: ``subprocess.CompletedProcess``
     """
     cmd_strlist = []
     cmd_logstr = []
     for member in cmd:
-        if isinstance(member, dict) and member.get("secret", False):
+        if isinstance(member, dict):
             cmd_strlist.append(member["item"])
-            cmd_logstr.append(REDACTED_OUT_SECRET)
+            if member.get("secret", False):
+                cmd_logstr.append(REDACTED_OUT_SECRET)
+            else:
+                cmd_logstr.append(member["item"])
         else:
             cmd_strlist.append(member)
             cmd_logstr.append(member)
-    try:
-        env = copy.deepcopy(os.environ)
-        if additional_env:
-            env.update(additional_env)
-        log.log(log_level, "%sRunning command '%s'",
-            "(DRYRUN) " if dry_run else "",
-            " ".join(cmd_logstr)
-        )
-        if dry_run:
-            result = subprocess.CompletedProcess(cmd_strlist, 0)
-        else:
-            stdout=subprocess.PIPE
-            stderr=subprocess.STDOUT if combine_out_err else subprocess.PIPE
-            result = subprocess.run(cmd_strlist, stdout=stdout, stderr=stderr, check=True, text=True, env=env)
-            log.log(log_level, "Command result:\n{}".format(
-                fmt_cmd_out_for_log(result, combine_out_err)
-            ))
-    except subprocess.CalledProcessError as e:
-        log.log(
-            log_level,
-            "Error in called process:\n{}".format(fmt_cmd_out_for_log(e, combine_out_err))
-        )
-        raise
-
-    return result
+    with logging_enabled(not sensitive_output):
+        try:
+            env = copy.deepcopy(os.environ)
+            if additional_env:
+                env.update(additional_env)
+            log.log(log_level, "%sRunning command '%s'",
+                "(DRYRUN) " if dry_run else "",
```

Comment:
Is it an issue that we'll get no output, even when in dry run ? It kinda removes the benefit of dry run no ?

Reply:
That's a good point. I'll make this only produce no output when this is a real run.